reconcile/utils/deadmanssnitch_api.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from typing import (
+    Optional,
+)
+
+import requests
+from typing import Any
+from pydantic import BaseModel
+
+BASE_URL = "https://api.deadmanssnitch.com/v1/snitches"
+class DeadManssnitchException(Exception):
+    pass
+
+class Snitch(BaseModel):
+    token: str
+    href: str
+    name: str
+    tags: list[str]
+    notes: str
+    status: str
+    check_in_url: str
+    interval: str
+    alert_type: str
+    alert_email: list[str]
+
+    def get_cluster_name(self) -> str:
+        return self.name.split(".")[1]
+
+    def to_dict(self)-> dict[str,Any]:
+        return self.dict(by_alias=True)
+
+class DeadMansSnitchApi:
+    def __init__(self, token: str, url: str = BASE_URL) -> None:
+        self.token = token
+        self.url = url
+
+    def get_snitches(self, tags: list[str]) -> list[Snitch]:
+        full_url = f"{self.url}?tags={','.join(tags)}"
+        logging.info("Getting snitches for tags:%s", tags)
+        response = requests.get(url=full_url, auth=(self.token, ""))
```

Comment:
since methods in this class can be called multiple times in one run, `requests.get/post/delete` will create a new session every time, should init a session during init, can do cleanup after usage, checkout `requests.Session()` usage, example [acs](https://github.com/app-sre/qontract-reconcile/blob/a804113c6645d60394086ccf7583ef24042fb6db/reconcile/utils/acs/base.py#L24)